users/user-guide-query/multi-stage-query/operator-types/leaf.md

```diff
@@ -0,0 +1,238 @@
+  mathematical operations are used), the multi-stage engine tries to keep the types (ie adding two integers will result 
+  in an integer).
+
+### Blocking nature
+One of the slight differences between the leaf and the normal single-stage engine is that the leaf engine tries to
+be not blocking.
+
+## Hints
+
+None
+
+## Stats
+### executionTimeMs
+Type: Long
+
+The summation of time spent by all threads executing the operator.
+This means that the wall time spent in the operation may be smaller that this value if the parallelism is larger than 1.
+
+### emittedRows
+Type: Long
+
+The number of groups emitted by the operator.
+
+### table
+Type: String
+
+The name of the table that is scanned. This is the name without the type suffix (so without `_REALTIME` or `_OFFLINE`).
+This is very useful to understand which table is being scanned by this leaf stage in case of complex queries.
+
+### numDocsScanned
+Type: Long
+
+Similar to the same stat in single-stage queries, this stat indicates the number of rows selected _after_ the 
+filter phase.
+
+If it is very high, that means the selectivity for the query is low and lots of rows need to be processed after the
+filtering. 
+Consider refining the filter to increase the selectivity of the query.
+
+### numEntriesScannedInFilter
+Type: Long
+
+Similar to the same stat in single-stage queries, this stat indicates the number of entries (aka scalar values) scanned 
+in the filtering phase of query execution.
+
+Can be larger than the total scanned doc count because of multiple filtering predicates or multi-value entries.
+Can also be smaller than the total scanned doc count if indexing is used for filtering.
+
+This along with `numEntriesScannedPostFilter` indicates where most of the time is spent during table scan processing. 
+If this value is high, enabling indexing for affected columns is a way to bring it down.
+Another option is to partition the data based on the dimension most heavily used in your filter queries.
+
+### numEntriesScannedPostFilter
+Type: Long
+
+Similar to the same stat in single-stage queries, this stat indicates the number of entries (aka scalar values) scanned
+after the filtering phase of query execution, ie. aggregation and/or group-by phases. 
+This is equivalent to `numDocScanned * number of projected columns`.
+
+This along with `numEntriesScannedInFilter` indicates where most of the time is spent during table scan processing.
+A high number for this means the selectivity is low (that is, Pinot needs to scan a lot of records to answer the query).
+If this is high, consider using star-tree index, given a regular index won't improve performance.
+
+### numSegmentsQueried
+Type: Integer
+
+Similar to the same stat in single-stage queries, this stat indicates the total number of segment queried for a query. 
+May be less than the total number of segments if the broker applies optimizations. <!-- Can this happen in V2? -->
```

Comment:
can we confirm this and update?

Reply:
seems from past exchanges, that is is not applied : https://startree.atlassian.net/browse/STP-3168 @Jackie-Jiang please help confirm